deeplake/core/vectorstore/vector_search/utils.py

```diff
@@ -99,11 +100,26 @@ def parse_search_args(**kwargs):
         raise ValueError(
             f"Either an `embedding`, `embedding_function`, `filter`, or `query` must be specified."
         )
+
+    if kwargs["embedding"] is not None and kwargs["embedding_function"]:
```

Comment:
Should it be `kwargs["embedding_data"]` here?

Reply:
I see. The idea was that embedding is the embedding data, but since now we have embedding_data which serves a bit different purpose, this error message is confusing.